### PR TITLE
Simplify logic. Adjust tests to work on name AND type.name

### PR DESCRIPTION
**Pull request by domfournier**
This modification should handle both vertex and cell association. We might want to split the merge data into two methods (one for each), but the idea is there.

### CODE REVIEW

**Comment by MatthieuCMira** [comment]
There is no pytest on this pull request.
If this logic works, it would be great, but I'd rather be sure it's 100% safe before changing the code.

**Comment by domfournier** [comment]
@sebhmg To address @MatthieuCMira comment, would you be against running the tests on PRs between branches?

File: tests/merger_test.py [hunk 1]
```diff
@@ -81,7 +80,7 @@ def test_merge_point_data_unique_entity(tmp_path):
         data.append(
```
Comment (MatthieuCMira) [inline tests/merger_test.py:80]: The first test functio is not pertinent anymore if we force data to have same name and entity type. 
Remove this one

File: tests/merger_test.py [hunk 2]
```diff
@@ -294,33 +269,16 @@ def test_merge_point_data_unique_entity_name_unique_name(tmp_path):
...
-                    }.items()
-                )
-            ).values()
-        )
-
-        np.testing.assert_almost_equal(
-            merged_data[0], np.hstack((nan_array, data[3].values))
-        )
-
         np.testing.assert_almost_equal(
```
Comment (MatthieuCMira) [inline tests/merger_test.py:272]: As we talk about, it's a bit of an assertion to consider the "first" data as the data to be merged. 
Would be safer to dispatch all the data in different data in this case no?

File: tests/merger_test.py [hunk 3]
```diff
@@ -294,33 +269,16 @@ def test_merge_point_data_unique_entity_name_unique_name(tmp_path):
...
-            merged_data[1], np.hstack((data[0].values, nan_array))
```
Comment (MatthieuCMira) [inline tests/merger_test.py:315]: What if the data with similar entity_type and  name are in the second object (or the nth object?
)

**Comment by sebhmg** [comment]
> @sebhmg To address @MatthieuCMira comment, would you be against running the tests on PRs between branches?

No problem. In practice PRs between intermediate branches do not happen often, thus running the tests will not cost to much against our quota. It might as well run the whole pre-commit checks. What do you think @domfournier ?

File: geoh5py/shared/merging/base.py [hunk 4]
```diff
@@ -30,150 +29,73 @@ class BaseMerger(ABC):
...
-
-        # prepare the attributes
-        unique_entity_types = list(
-            dict.fromkeys(
-                [entity_type[:nb_keys] for entity_type in unique_entity_types]  # type: ignore
+                # Increment label in case of duplicate match
+                if label in data_dict:
+                    values = data_dict[label].values
+                    if isinstance(values, np.ndarray) and ~np.all(
+                        np.isnan(values[start:end])
```
Comment (MatthieuCMira) [inline geoh5py/shared/merging/base.py:63]: That's a good idea.
However, how to ensure the user want to merge his first data, not the second one.

I agree it's stupid, so we don't really have to care about it programmatically,  but that's a possible scenario. 
I would just pass a warning there "Just the first data encountered was added"

**Review by MatthieuCMira — COMMENTED** [review]
I really like the overall logic, much more elegant. 

Just passing a warning in the case where data are "overlapping", precising the data merged might not be the one wanted.

**Comment by MatthieuCMira** [comment]
merge dom's data merger solution.
Need to adapt some tests

**Pull request merged**